From 0940e8a7e89f58686a7c0275cf2331c4e9759d78 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lu=C3=ADs=20Pereira?= <luis.artur.pereira@gmail.com>
Date: Mon, 2 Feb 2015 11:33:44 -0800
Subject: [PATCH 08/15] QIconLoader: replace an inefficient QList with a
 QVector

QIconDirInfo is larger than a pointer, so holding it in a
QList is horribly inefficient.

Fix by holding it in a QVector instead.

Port of commit
https://qt.gitorious.org/qt/qtbase/commits/87ccab8bd690bfb6644175ab4956a57838b48a04
---
 qiconfix/qiconloader.cpp | 2 +-
 qiconfix/qiconloader_p.h | 5 +++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/qiconfix/qiconloader.cpp b/qiconfix/qiconloader.cpp
index 956d57a..8ae43ee 100644
--- a/qiconfix/qiconloader.cpp
+++ b/qiconfix/qiconloader.cpp
@@ -288,7 +288,7 @@ QThemeIconEntries QIconLoader::findIconHelper(const QString &themeName,
     }
 
     QStringList contentDirs = theme.contentDirs();
-    QList<QIconDirInfo> subDirs = theme.keyList();
+    const QVector<QIconDirInfo> subDirs = theme.keyList();
 
     const QString svgext(QLatin1String(".svg"));
     const QString pngext(QLatin1String(".png"));
diff --git a/qiconfix/qiconloader_p.h b/qiconfix/qiconloader_p.h
index 73fdcd4..dc74b3e 100644
--- a/qiconfix/qiconloader_p.h
+++ b/qiconfix/qiconloader_p.h
@@ -62,6 +62,7 @@
 //#include <private/qicon_p.h>
 //#include <private/qfactoryloader_p.h>
 #include <QtCore/QHash>
+#include <QtCore/QVector>
 #include <QtCore/QTypeInfo>
 
 //QT_BEGIN_NAMESPACE
@@ -148,7 +149,7 @@ public:
     QIconTheme(const QString &name);
     QIconTheme() : m_valid(false) {}
     QStringList parents() { return m_parents; }
-    QList <QIconDirInfo> keyList() { return m_keyList; }
+    QVector <QIconDirInfo> keyList() { return m_keyList; }
     QString contentDir() { return m_contentDir; }
     QStringList contentDirs() { return m_contentDirs; }
     bool isValid() { return m_valid; }
@@ -156,7 +157,7 @@ public:
 private:
     QString m_contentDir;
     QStringList m_contentDirs;
-    QList <QIconDirInfo> m_keyList;
+    QVector <QIconDirInfo> m_keyList;
     QStringList m_parents;
     bool m_valid;
 };
-- 
1.9.0

